### PR TITLE
DAH-1126 invalid address validation with help message

### PR DESCRIPTION
**Pull request by christine-sfg**
DAH-1126 (screenshots in the jira)

-Add back validations for invalid addresses. Reverts https://github.com/SFDigitalServices/sf-dahlia-web/pull/1532
-Display a helpful error message so users are not blocked from applying. The error message includes a `mailto` which pre-populates an email with relevant info to help resolve the issue
-Continue preventing po boxes

### CODE REVIEW

File: spec/requests/api/v1/address_validation_spec.rb [hunk 1]
```diff
@@ -76,7 +76,7 @@
     expect(json['address']['verifications']['delivery']['success']).to eq(true)
   end
 
-  it 'allows invalid addresses to continue through' do
+  it 'validates invalid address with success == false' do
     VCR.use_cassette('address_validation/invalid') do
```
Comment (jacksongonzales) [inline spec/requests/api/v1/address_validation_spec.rb:80]: smh this stuff is so rusty that it uses VCR cassettes 🙄

Comment (christine-sfg) [reply spec/requests/api/v1/address_validation_spec.rb:80]: 😂

**Review by jacksongonzales — APPROVED** [review]
this looks great! the ruby code scans for me, and testing passes all the workflows laid out in the review instructions 🎉

**Comment by emilyjablonski** [comment]
LGTM! From an AC perspective just curious why we don't validate mailing addresses?

**Comment by christine-sfg** [comment]
> LGTM! From an AC perspective just curious why we don't validate mailing addresses?

Thanks! That's a great question. I'm guessing it's because home address is essential to qualifying for certain preferences. Mailing address is something we should look into adding for the rewrite.

**Pull request merged**